### DISCUSSION TITLE
Array nodal bcs for eigenproblem

### DISCUSSION THREAD

**Discussion by xieeeect** [idaholab/moose#28455] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,

I'm working on a array nodal boundary condition for eigenvalue problems, it describes like: M * U = 0 , M is an n*n matrix, and U is an array variable. However, I didn't find any BCS about it, and when developing a new bcs with 'ArrayNodalBC', I got an error: 
          "Invalid NodalBC for eigenvalue problems, please use homogeneous (array) Dirichlet."

Thanks for your time,

xieeeect

**Comment by GiudGiud** [comment]
@YaqiWang I think we have had this conversation before. 
Do you recall the conclusions? What kind of BCs can be used for eigenvalue problems?

**Reply by GiudGiud** [reply]
or @lindsayad I think you know exactly what's up  here considering your latest work

**Reply by lindsayad** [reply, marked as answer]
The short answer is that I do not know the theory well enough to say what BCs can theoretically be used for eigenvalue problems. I can only speak to the implementation which is that we only allow homogeneous BCs in MOOSE.

From the libMesh side we now allow solving eigenproblems with arbitrary constraints in which we condense out the constrained degrees of freedom when performing the eigensolve. We then enforce the constraints post-eigensolve on the requested eigenvector

**Reply by xieeeect** [reply]
Thanks for the answer, I have a general understanding of the rules for handling BCs for eigenvalue problems, and next I will study the idea you said.  : )
